xdem/spstats.py

```diff
@@ -0,0 +1,507 @@
+"""
+xdem.spstats provides tools to use spatial statistics for elevation change data
+"""
+from __future__ import annotations
+import os
+import numpy as np
+import math as m
+import skgstat as skg
+from scipy import integrate
+import multiprocessing as mp
+import pandas as pd
+import ogr
+from functools import partial
+
+def get_empirical_variogram(dh: np.ndarray,coords: np.ndarray, **kwargs) -> pd.DataFrame:
+    """
+    Get empirical variogram from skgstat.Variogram object
+
+    :param dh: elevation differences
+    :param coords: coordinates
+    :return: empirical variogram (variance, lags, counts)
+
+    """
+
+    # deriving empirical variogram variance, bin, and count
+    try:
+        V = skg.Variogram(coordinates=coords, values=dh, normalize=False, **kwargs)
+        # return V.to_DataFrame()
+
+        exp = V.experimental
+        bins = V.bins
+        count = np.zeros(V.n_lags)
+        tmp_count = np.fromiter((g.size for g in V.lag_classes()), dtype=int)
+        count[0:len(tmp_count)] = tmp_count
+
+    # there are still some exceptions not well handled by skgstat
+    except:
+        if 'n_lags' in kwargs.keys():
```

Comment:
This could be shortened to:
```python
n_lags = kwargs.get('n_lags') or 10
```
If `kwargs.get('n_lags')` returns None, it defaults to 10!

Reply:
Nice tip! Love learning those =D